### PR TITLE
ARTEMIS-565 Replace json.org with javax.json

### PR DESCRIPTION
**Pull request by johnament**
Javax.json is a newer JSR, but has an ASF compliant version, is pretty close to the original JSON.org API and will support a standard annotation based JSON-B solution at some point soon.

This is for discussion purposes only at this point.

### CODE REVIEW

**Comment by mtaylor** [comment]
@johnament +1 on this.  TBH I don't really mind which library we use as long as it's Apache compliant and doesn't have silly phrases in the license :P.  This looks good to me.

**Comment by puntogil** [comment]
hi
thanks for your work
is possible remove from LICENSE file the json.org (license) referencies
regards

**Comment by johnament** [comment]
@puntogil yep, that's why i'm not saying this is ready.

@mtaylor Ok, I'll clean it up.

Thanks for looking at this guys!

**Comment by johnament** [comment]
Ok, the changes should be good.  I manually ran the integration tests that were touched, and fixed a bunch of issues identified there.  

I'm going to create a ticket to create more lower level json tests.  They're a good way to verify contracts in a situation like this.  If I were starting this over, I would have added the contracts first.

**Comment by johnament** [comment]
I'm not sure why Jenkins didn't write the result back, but the build did pass in Jenkins.

**Comment by clebertsuconic** [comment]
@johnament this looks good to me. Are you done here? Can I merge it.

**Comment by clebertsuconic** [comment]
you can't run a server. There is some changes needed on the release packaging:

```
Exception in thread "main" java.lang.NoClassDefFoundError: javax/json/JsonValue
	at org.apache.activemq.artemis.uri.schema.connector.TCPTransportConfigurationSchema.getTransportConfigurations(TCPTransportConfigurationSchema.java:68)
	at org.apache.activemq.artemis.uri.schema.serverLocator.TCPServerLocatorSchema.internalNewObject(TCPServerLocatorSchema.java:43)
	at org.apache.activemq.artemis.uri.schema.serverLocator.TCPServerLocatorSchema.internalNewObject(TCPServerLocatorSchema.java:33)
	at org.apache.activemq.artemis.utils.uri.URISchema.newObject(URISchema.java:87)
	at org.apache.activemq.artemis.utils.uri.URISchema.newObject(URISchema.java:30)
	at org.apache.activemq.artemis.utils.uri.URIFactory.newObject(URIFactory.java:59)
	at org.apache.activemq.artemis.core.client.impl.ServerLocatorImpl.newLocator(ServerLocatorImpl.java:413)
	at org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.<init>(ActiveMQConnectionFactory.java:179)
	at org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.<init>(ActiveMQConnectionFactory.java:199)
	at org.apache.activemq.artemis.cli.commands.Producer.execute(Producer.java:52)
	at org.apache.activemq.artemis.cli.Artemis.internalExecute(Artemis.java:110)
	at org.apache.activemq.artemis.cli.Artemis.execute(Artemis.java:72)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.apache.activemq.artemis.boot.Artemis.execute(Artemis.java:122)
	at org.apache.activemq.artemis.boot.Artemis.main(Artemis.java:45)
Caused by: java.lang.ClassNotFoundException: javax.json.JsonValue
```

**Comment by johnament** [comment]
Ah good catch.  I can fix that, probably some assembly issues.  I'm assuming shipping johnzon as the default is fine?

**Comment by clebertsuconic** [comment]
Looks fine to me.

I reversed engineered the code and I'm not sure how you specify the JSON provider. it seems you need something on the CLASSINFO?

Also, I just realized the main pom doesn't specify the licenses. a small thing but since you're changing it, can you also do it? will comment at the pom to make it easier to find it.

File: pom.xml [hunk 1]
```diff
@@ -436,6 +436,17 @@
            <version>${slf4j.version}</version>
            <!-- License: MIT -->
          </dependency>
+         <!-- json -->
+         <dependency>
+            <groupId>org.apache.geronimo.specs</groupId>
+            <artifactId>geronimo-json_1.0_spec</artifactId>
+            <version>${json-p.spec.version}</version>
```
Comment (clebertsuconic) [inline pom.xml]: Do you mind mentioning Apache license here, just to keep up the list?

File: pom.xml [hunk 2]
```diff
@@ -436,6 +436,17 @@
...
+         </dependency>
+         <dependency>
+            <groupId>org.apache.johnzon</groupId>
+            <artifactId>johnzon-core</artifactId>
+            <version>${johnzon.version}</version>
```
Comment (clebertsuconic) [inline pom.xml]: Do you mind mentioning Apache license here, just to keep up the list?

Comment (clebertsuconic) [inline pom.xml]: hmmm... those are org.apache.. duh.. not really needed :)

**Comment by clebertsuconic** [comment]
@johnament how users doing embedded will use json? anything special they have to do? or just the maven dependencies would catch it?

**Comment by johnament** [comment]
@clebertsuconic yes, once i fix the distribution the johnzon one will ship by default.

**Comment by clebertsuconic** [comment]
@johnament I was thinking about embedded cases (using the library direct), like wildfly.

We can fix whatever needs on wildfly for next versions.. so I was really thinking about other cases.. I will look through the examples.

Maybe you could also check it by running artemis-distribution/src/test/scripts/run-examples.sh  after building the server.

**Comment by johnament** [comment]
@clebertsuconic oh, for wildfly, you'll need to add dependencies on `javax.json.api` and `org.glassfish.javax.json` modules.

As far as plain library usage, the dependency tree should come in.  If you want to plugin your own, just exclude.

**Comment by clebertsuconic** [comment]
@johnament  cool.. thanks a lot.
And sorry for my delay on this. was in vacations last week. :)

**Comment by johnament** [comment]
No problems @clebertsuconic hope it went well.

I pushed up fixes, one oddity is that when starting the broker I get this on the console...

```
20:50:26,422 INFO  [org.apache.activemq.artemis.core.server] Checking...
20:50:28,425 INFO  [org.apache.activemq.artemis.core.server] Checking...
20:50:30,428 INFO  [org.apache.activemq.artemis.core.server] Checking...
```

Not sure if its related to my changes, or something else.

**Comment by johnament** [comment]
I just checked.  its happening with my changes.  I can't find this log message anywhere though.

**Comment by jbertram** [comment]
You need to rebase.  That issue was resolved last week I believe.

**Comment by clebertsuconic** [comment]
I checked your commit and it was ok.

(I always use this script to merge PRs which will always rebase them, that's probably why it was ok)...
https://github.com/apache/activemq-artemis/blob/master/scripts/merge-PR.sh

no need to rebase on that case

**Comment by clebertsuconic** [comment]
I included 2 lines change features.xml towards the OSGI bundle and I squashed into your commit. I hope you won't mind. It was easier this way.

**Comment by clebertsuconic** [comment]
@johnament can you assign this JIRA to yourself? I couldn't find you (don't know why):

https://issues.apache.org/jira/browse/ARTEMIS-565

**Comment by johnament** [comment]
I pinged the PMC to add me as contributor.  That will allow JIRA assignment.

I must have had something cached, as this was after I rebased.

**Pull request closed without merging**